misc/sieve_of_eratosthenes.c

```diff
@@ -0,0 +1,73 @@
+/**
+ * @file
+ * @brief Get list of prime numbers using [Sieve of
+ * Eratosthenes](https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes)
+ * @details
+ * Sieve of Eratosthenes is an algorithm that finds all the primes
+ * between 2 and N.
+ *
+ * Time Complexity  : \f$O(N \cdot\log \log N)\f$
+ * <br/>Space Complexity : \f$O(N)\f$
+ *
+ */
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
```

Comment:
Provide a brief description of the headers (what do they add).
```suggestion
#include <assert.h>  /// for assert
#include <stdbool.h> ///
#include <stdio.h>   ///
#include <stdlib.h>  ///
#include <string.h>  ///
```